single_cell/getting-started-with-hca.ipynb

```diff
@@ -0,0 +1,492 @@
+    "    \n",
+    "- URL endpoint for the HCA catalog\n",
+    "- Where to save downloaded files\n",
+    "- An example project's UUID to download"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "residential-techno",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "CATALOG = 'dcp30'\n",
+    "ENDPOINT_URL = f'https://service.azul.data.humancellatlas.org/index/projects'\n",
+    "\n",
+    "HCA_EXAMPLES_DIR=os.path.expanduser('~/terra-tutorials/hca')\n",
+    "OUTPUT_DIR = os.path.join(HCA_EXAMPLES_DIR, 'data')\n",
+    "\n",
+    "!mkdir -p \"{OUTPUT_DIR}\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6674304b-1ee6-430d-9238-36992281140b",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "### Create utility routines"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7b82cbfe-7f7b-4760-8722-9e810337df0f",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "#### download_file \n",
+    "\n",
+    "Downloads the content of the specified URL to a local output path,\n",
+    "while displaying a progress bar."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "painful-monitoring",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def download_file(url, output_path):\n",
+    "    # Start the request stream\n",
+    "    response = requests.get(url, stream=True)\n",
+    "    response.raise_for_status()\n",
+    "\n",
+    "    # Get the content length so the progress bar can display accurate progress\n",
+    "    total = int(response.headers.get('content-length', 0))\n",
+    "    print(f'Downloading to: {output_path}', flush=True)\n",
+    "    \n",
+    "    # Fetch the content in chunks, updating the progress bar\n",
+    "    with open(output_path, 'wb') as f:\n",
+    "        with tqdm(total=total, unit='B', unit_scale=True, unit_divisor=1024) as bar:\n",
+    "            for chunk in response.iter_content(chunk_size=1024):\n",
+    "                size = f.write(chunk)\n",
+    "                bar.update(size)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "49efcf24-7e1f-411c-8fc2-5628104daf2d",
+   "metadata": {},
+   "source": [
+    "#### fetch_projects \n",
+    "\n",
+    "Fetch the list of projects in the HCA catalog."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e60613bb-4f25-4afe-a45a-ffc419a1762c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def fetch_projects(catalog: str, max_projects: int) -> list:\n",
+    "\n",
+    "    # Set up request parameters\n",
+    "    params = {\n",
+    "      'catalog': catalog,\n",
+    "      'size': max_projects,\n",
+    "      'sort': 'projectTitle',\n",
+    "      'order': 'asc'\n",
+    "    }\n",
+    "    \n",
+    "    # Fetch the Catalog\n",
+    "    response = requests.get(ENDPOINT_URL, params=params)\n",
+    "    response.raise_for_status()\n",
+    "    \n",
+    "    return response"
```

Comment:
IMO this should be `return response.json()` since the type hint says this function returns a list